### PR TITLE
CNFT1-2806 Clear all

### PR DESCRIPTION
**Pull request by jlee-enquizit**
## Description

Enabling 'Clear all' button in Patient Search, Investigation Search, Lab report search.

## Tickets

* [CNFT1-2806](https://cdc-nbs.atlassian.net/browse/CNFT1-2806)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT1-2806]: https://cdc-nbs.atlassian.net/browse/CNFT1-2806?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by stevegsa — APPROVED** [review]

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
The form is already being reset when the search status is being set to `waiting` with a reset.  When "Clear all" is clicked the search results should be cleared, that is not happening any more.  

There are two issues causing this; 

1. The form value is not being passed to down to the component that renders these fields.  
2. The resetting of the form happens when the search status is changed to `waiting`, clicking the "Clear all" button before clicking "Search" the status is going from `waiting` to `waiting` which is not considered a change in react.

**Comment by jlee-enquizit** [comment]
> The form is already being reset when the search status is being set to `waiting` with a reset. When "Clear all" is clicked the search results should be cleared, that is not happening any more.
> 
> There are two issues causing this;
> 
> 1. The form value is not being passed to down to the component that renders these fields.
> 2. The resetting of the form happens when the search status is changed to `waiting`, clicking the "Clear all" button before clicking "Search" the status is going from `waiting` to `waiting` which is not considered a change in react.

got it!

File: apps/modernization-ui/src/apps/search/useSearch.ts [hunk 1]
```diff
@@ -122,9 +127,10 @@ const useSearch = <C, A, R>({ transformer, resultResolver, termResolver }: Setti
             searchResults.search();
         } else if (state.status === 'completed') {
             searchResults.complete(state.results.terms);
-        } else if (state.status === 'waiting') {
+        } else if (state.status === 'resetting') {
             pageReset();
             searchResults.reset();
+            dispatch({ type: 'wait' });
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/useSearch.ts:133]: Added a status to let the callers know that search is resetting

File: apps/modernization-ui/src/apps/search/investigation/InvestigationSearch.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ const InvestigationSearch = () => {
     const { terms } = useSearchResultDisplay();
 
     useEffect(() => {
-        if (status === 'waiting') {
+        if (status === 'resetting') {
             form.reset();
         }
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/search/investigation/InvestigationSearch.tsx:25]: The form only needs to reset when search is resetting.

**Review by stevegsa — APPROVED** [review]

**Review by adamloup-enquizit — APPROVED** [review]

**Pull request merged**